hack/scripts/quick_reset.sh

```diff
@@ -31,8 +31,12 @@ fi
 # Validate that we got a valid block device to cleanup
 validateBlockDevice
 
+echo "Calling blkdiscard (failure is permitted -- some devices do not support discard)"
+ionice -c 3 blkdiscard -f $LOCAL_PV_BLKDEVICE || true
+
 echo "Calling mkfs"
-ionice -c 3 mkfs -F $LOCAL_PV_BLKDEVICE
+# mkfs.xfs is preferred because default mkfs has 16TB limit (https://access.redhat.com/solutions/66689)
+ionice -c 3 mkfs.xfs -f $LOCAL_PV_BLKDEVICE
```

Comment:
* `blkdiscard` above marks all blocks as unused, so lower storage layers may re-use the blocks. This is just to be nice to the underlying device and may not discard anything on devices that don't support it.
* `wipefs` below destroys headers of *known* filesystems, partition tables and similar things, so the device appears as it does not have any.

So why do we need `mkfs.xfs`, that destroys the first 512 bytes (overwrites them with xfs superblock) + puts xfs metadata on the disk and discards all other blocks (if supported by the device)? One reason could be to destroy header of unknown data on the device, when it is used as a raw block and e.g. a database is stored there without any filesystem. We should either add a comment about this use case or replace it with something better (`dd` the first block?)

Reply:
For discard support detection, would [this](https://unix.stackexchange.com/a/668323) be helpful? Seems very simple to implement.

I agree that doing a `mkfs` at all seems a bit weird. A simple `dd` should do the trick and not require a `wipefs` call anymore. I saw an implementation for this [upstream](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/deployment/docker/scripts/dd_zero.sh), though I'm sure you've also already seen it.

Reply:
I've just double-checked the behavior of mkfs.xfs and observed that it always discard all blocks from the device (if discard is supported).

Reply:
@Alveel, full `dd` is too heavy (would take too long), and using mkfs gives us some trade off clearing at least some amount of data. This PR simply replaces mkfs.ext2 with mkfs.xfs. I agree that mkfs seems a bit weird, but it corresponds to upstream and it's enough to fix OCPBUGS-22250.

Reply:
Of course, I think you'd only want to `dd` the first few blocks. The upstream script allows for that :)

Regardless, happy to see this bug fixed this quickly.